### DISCUSSION TITLE
Execution Sequence Alloy MVC

### DISCUSSION THREAD

**Discussion by beppo-ivel** [appcelerator/titanium_mobile#13040] [Q&A] [answered]
If I run the following example, then  `Ti.API.error("hello");` is print out before `Ti.API.error(JSON.stringify(success));`. I expected that tab1.js controller is loaded after the call of `$.index.open();`
controllers/index.js
```
aCustomPromise()
.then(function(success){
	Ti.API.error(JSON.stringify(success));
	$.index.open();
})
.catch(function(error){
	Ti.API.error(JSON.stringify(error));
	$.index.close();
});

```
views/index.xml
```
<Alloy>
	<TabGroup title="MyApp">
           <Require src="tab1" />
	</TabGroup>
</Alloy>
```
views/tab1.xml
```
<Alloy>
  <Tab id="tab1" title="Tab">
    <Window id="window1" title="Window 1" onClose="cleanup">
    </Window>
  </Tab>
</Alloy>
```

controllers/tab1.js
```
(function() {
  Ti.API.error("hello");
})();
function cleanup() {
    $.destroy();
}

```

**Comment by ewanharris** [comment, marked as answer]
@beppo-ivel, any code at the top level of the controller will be executed when `Alloy.createController` is called. If you open up the compile Classic source at `Resources/<platform>/alloy/controllers/index.js`, you can see that the tab1 instance is created (executing the top level code), then the TabGroup is created, then `aCustomPromise` is called.

If you want to delay something until tab1 is actually open, you can use the `onOpen` event handler. That should give you the flow I believe you're expecting
